app/assets/javascripts/member-facing/backbone/fundraiser.js

```diff
@@ -221,29 +283,56 @@ const Fundraiser = Backbone.View.extend(_.extend(
     window.open(url);
   },
 
-  submitDonation (nonce) {
+  submitDonation(nonce) {
     let data = this.donationData();
     data.payment_method_nonce = nonce;
-    $.post(`/api/payment/braintree/pages/${this.pageId}/transaction`, data).
-      done(this.transactionSuccess.bind(this)).
-      error(this.transactionFailed.bind(this));
-    if(this.directDebitOpened){
+    $.post(`/api/payment/braintree/pages/${this.pageId}/transaction`, data)
+      .then(this.onSubmission.bind(this))
+      .then(this.transactionSuccess.bind(this), this.transactionFailed.bind(this));
+
+    if (this.directDebitOpened) {
       $.publish('direct_debit:donated_via_other');
     }
   },
 
-  transactionSuccess(data, status) {
-    let hasCallbackFunction = (typeof this.submissionCallback === 'function');
-    if (hasCallbackFunction) {
+  onSubmission(data, status) {
+    if (this.submissionCallback) {
       this.submissionCallback(data, status);
     }
-    if (this.followUpUrl) {
-      this.redirectTo(this.followUpUrl);
-    }
-    if (!this.followUpUrl && !hasCallbackFunction) {
-      // only do this option if no redirect or callback supplied
-      alert(I18n.t('fundraiser.thank_you'));
-    }
+  },
+
+  submitOneClick(event) {
+    event.preventDefault();
+    const url = `/api/payment/braintree/pages/${this.pageId}/one_click`;
+    this.disableOneClickButton();
+    $.post(url, this.oneClickDonationData())
+      .then(this.onSubmission.bind(this))
+      .then(this.onOneClickSuccess.bind(this), this.onOneClickFailed.bind(this));
+  },
+
+  onOneClickSuccess() {
+    if ( this.memberShouldRegister() )
+      this.followRedirect(this.registrationPath(window.champaign.personalization.member.email));
+    else
+      this.followRedirect(this.followUpUrl);
+  },
+
+  transactionSuccess() {
+    const user = this.serializeUserForm();
+    let url = this.followUpUrl;
+
+    if ( this.memberShouldRegister() )
```

Comment:
I'm happy to work toward improving it, but the fact that it doesn't pass the linter doesn't mean we shouldn't stay internally consistent when we notice it